go.mod

```diff
@@ -1,35 +1,44 @@
 module github.com/hashicorp/vault-plugin-secrets-azure
 
-go 1.19
+go 1.18
 
 require (
 	github.com/Azure/azure-sdk-for-go v68.0.0+incompatible
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.8.0
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.4.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources v1.1.1
 	github.com/Azure/go-autorest/autorest v0.11.29
 	github.com/Azure/go-autorest/autorest/azure/auth v0.5.12
 	github.com/Azure/go-autorest/autorest/date v0.3.0
 	github.com/Azure/go-autorest/autorest/to v0.4.0
 	github.com/go-test/deep v1.1.0
 	github.com/golang/mock v1.6.0
+	github.com/google/uuid v1.3.1
```

Comment:
ah, didn't realize we'd be importing this library. Can we try to use `go-uuid` instead?